indexer/indexing.py

```diff
@@ -92,66 +107,106 @@ async def create_pg_vector_index_if_not_exists(self):
             port=self.db_port,
         )
         try:
-            # Create an asynchronous cursor object to interact with the database
             async with connection.transaction():
-                # Execute the alter table query
                 await connection.execute(
                     "ALTER TABLE langchain_pg_embedding ALTER COLUMN embedding TYPE vector(1536)"
                 )
                 await connection.execute(
                     "CREATE INDEX IF NOT EXISTS langchain_embeddings_hnsw ON langchain_pg_embedding USING hnsw (embedding vector_cosine_ops)"
                 )
         finally:
-            # Close the connection
             await connection.close()
 
-    async def index(self, indexer_input: IndexerInput):
+    async def index(self, indexer_input: Indexer):
+        source_files = []
         source_chunks = []
         counter = 0
-        for file in indexer_input.files:
-            file_path = os.path.join(os.environ["DOCUMENT_LOCAL_STORAGE_PATH"], file)
-            print("file_path", file_path)
-            content = await self.text_converter.textify(file_path)
-            for chunk in self.splitter.split_text(content):
-                new_metadata = {
-                    "chunk-id": str(counter),
-                    "document_name": file,
-                }
-                source_chunks.append(
-                    Document(page_content=chunk, metadata=new_metadata)
+        for file_path in indexer_input.files:
+            tmp_file_path = await storage._download_file_to_temp_storage(
```

Comment:
@KaranrajM Can we remove call to _download_file_to_temp_storage? It is supposed to be a private method.